frank-flow/src/frontend/src/app/shared/services/layout.service.ts

```diff
@@ -9,20 +9,43 @@ import { BehaviorSubject } from 'rxjs';
 export class LayoutService {
   private nodesSubject = new BehaviorSubject<Map<string, Node>>(new Map());
   public nodesObservable = this.nodesSubject.asObservable();
+
   private readonly LEFT_MARGIN = 100;
   private readonly TOP_MARGIN = 100;
   private readonly NODE_HEIGHT = 100;
   private readonly NODE_WIDTH = 200;
   private readonly rows = new Map<string, number>();
+  private rowStartingPointPerColumn: number[] = []
 
   constructor(private nodeService: NodeService) {}
 
   createLayout(nodeMap: Map<string, Node>): void {
     this.rows.clear();
+    this.rowStartingPointPerColumn = [];
+    this.determineStartingPointPerRow(nodeMap);
     this.addNodesToCanvas(nodeMap);
     this.nodesSubject.next(nodeMap);
   }
 
+  determineStartingPointPerRow(nodeMap: Map<string, Node>) {
+    for (const node of nodeMap.values()) {
+      let row: number
+      let column: number
+      if (/Receiver/g.test(node.getType())) {
```

Comment:
Does this need to be regex or can `string.includes()` be used instead?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes

Performance wise its kind of all over the place on sources about what is faster but maybe for simplicity using `.includes()` might be better

Reply:
I think the linter did it, I'll change it to `includes()`